beacon_node/execution_layer/src/engine_api/http.rs

```diff
@@ -48,6 +49,480 @@ pub const BUILDER_GET_PAYLOAD_HEADER_TIMEOUT: Duration = Duration::from_secs(2);
+            let result = self
+                .call(address, DEPOSIT_COUNT_FN_SIGNATURE, block_number, timeout)
+                .await?;
+            match result {
+                None => Err("Deposit root response was none".to_string()),
+                Some(bytes) => {
+                    if bytes.is_empty() {
+                        Ok(None)
+                    } else if bytes.len() == DEPOSIT_COUNT_RESPONSE_BYTES {
+                        let mut array = [0; 8];
+                        array.copy_from_slice(&bytes[32 + 32..32 + 32 + 8]);
+                        Ok(Some(u64::from_le_bytes(array)))
+                    } else {
+                        Err(format!(
+                            "Deposit count response was not {} bytes: {:?}",
+                            DEPOSIT_COUNT_RESPONSE_BYTES, bytes
+                        ))
+                    }
+                }
+            }
+        }
+
+        /// Returns the value of the `get_hash_tree_root()` call at the given `block_number`.
+        ///
+        /// Assumes that the `address` has the same ABI as the eth2 deposit contract.
+        pub async fn get_deposit_root(
+            &self,
+            address: &str,
+            block_number: u64,
+            timeout: Duration,
+        ) -> Result<Option<Hash256>, String> {
+            let result = self
+                .call(address, DEPOSIT_ROOT_FN_SIGNATURE, block_number, timeout)
+                .await?;
+            match result {
+                None => Err("Deposit root response was none".to_string()),
+                Some(bytes) => {
+                    if bytes.is_empty() {
+                        Ok(None)
+                    } else if bytes.len() == DEPOSIT_ROOT_BYTES {
+                        Ok(Some(Hash256::from_slice(&bytes)))
+                    } else {
+                        Err(format!(
+                            "Deposit root response was not {} bytes: {:?}",
+                            DEPOSIT_ROOT_BYTES, bytes
+                        ))
+                    }
+                }
+            }
+        }
+
+        /// Performs a instant, no-transaction call to the contract `address` with the given `0x`-prefixed
+        /// `hex_data`.
+        ///
+        /// Returns bytes, if any.
+        async fn call(
+            &self,
+            address: &str,
+            hex_data: &str,
+            block_number: u64,
+            timeout: Duration,
+        ) -> Result<Option<Vec<u8>>, String> {
+            let params = json! ([
+                {
+                    "to": address,
+                    "data": hex_data,
+                },
+                format!("0x{:x}", block_number)
+            ]);
+
+            let response: String = self
+                .rpc_request("eth_call", params, timeout)
+                .await
+                .map_err(|e| format!("eth_call call failed {:?}", e))?;
+
+            Ok(Some(hex_to_bytes(&response)?))
```

Comment:
```suggestion
            let response: Option<String> = self
                .rpc_request("eth_call", params, timeout)
                .await
                .map_err(|e| format!("eth_call call failed {:?}", e))?;

            response.map(|s| hex_to_bytes(&s)).transpose()
```

This ensures we still allow `eth_call` to return `None`. I don't think we actually use `None` anywhere, but this change will keep the function signature honest. Here's the old code:

https://github.com/sigp/lighthouse/blob/cfd26d25e039ed85b67a0b88f02ef633d64b9119/beacon_node/eth1/src/http.rs#L328-L329